### DISCUSSION TITLE
Regarding #define macros in fips32. for latest openssl

### DISCUSSION THREAD

**Discussion by pb1234** [openssl/openssl#24045] [Q&A]
Hi ,
Our old code was using OpenSSL 1.0.2 q and it was using fips32, our code using these macros as below which were defined in fips32 of older version.
CRYPTO_LOCK_SSL_CTX
CRYPTO_LOCK_SSL_SESSION
CRYPTO_add(&a->references, 1, CRYPTO_LOCK_SSL_SESSION); //sample code for reference.

So now we updated to more recent version of OpenSSL 3.x, and these macros are not defined in fips32 now, but these are there in fips64, what can we do here, we tried just locally define in the source files but that is throwing linker errors.

**Comment by slontis** [comment]
It uses a new threading model API.
Which uses XXX_up_ref() calls for refcounting.. (Where XXX is a name such as SSL_SESSION).
See the 8 year old commits here (https://github.com/openssl/openssl/commit/16203f7b71bd343550f89f266eaf9fb9693f6148).
Refer to the documentation for SSL_SESSION_up_ref() for more info.
